godot-core/src/builtin/meta/godot_compat/impls.rs

```diff
@@ -0,0 +1,248 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::builtin::meta::GodotType;
+
+use super::*;
+
+impl<T: GodotCompatible> GodotCompatible for Option<T>
+where
+    Option<T::Via>: GodotType,
+{
+    type Via = Option<T::Via>;
+}
+
+impl<T: ToGodot> ToGodot for Option<T>
+where
+    Option<T::Via>: GodotType,
+{
+    fn to_godot(&self) -> Self::Via {
+        self.as_ref().map(ToGodot::to_godot)
+    }
+
+    fn into_godot(self) -> Self::Via {
+        self.map(ToGodot::into_godot)
+    }
+}
+
+impl<T: FromGodot> FromGodot for Option<T>
+where
+    Option<T::Via>: GodotType,
+{
+    fn try_from_godot(via: Self::Via) -> Option<Self> {
+        match via {
+            Some(via) => T::try_from_godot(via).map(Some),
+            None => Some(None),
+        }
+    }
+
+    fn from_godot(via: Self::Via) -> Self {
+        via.map(T::from_godot)
+    }
+}
+
+impl GodotCompatible for sys::VariantType {
+    type Via = i32;
+}
+
+impl ToGodot for sys::VariantType {
+    fn to_godot(&self) -> Self::Via {
+        *self as i32
+    }
+
+    fn into_godot(self) -> Self::Via {
+        self as i32
+    }
+}
+
+impl FromGodot for sys::VariantType {
+    fn try_from_godot(via: Self::Via) -> Option<Self> {
+        Some(Self::from_sys(via as sys::GDExtensionVariantType))
+    }
+}
+
+impl GodotCompatible for sys::VariantOperator {
+    type Via = i32;
+}
+
+impl ToGodot for sys::VariantOperator {
+    fn to_godot(&self) -> Self::Via {
+        *self as i32
+    }
+
+    fn into_godot(self) -> Self::Via {
+        self as i32
+    }
+}
+
+impl FromGodot for sys::VariantOperator {
+    fn try_from_godot(via: Self::Via) -> Option<Self> {
+        Some(Self::from_sys(via as sys::GDExtensionVariantOperator))
+    }
+}
+
+impl<T> GodotCompatible for *mut T {
+    type Via = i64;
+}
+
+impl<T> ToGodot for *mut T {
+    fn to_godot(&self) -> Self::Via {
+        *self as i64
+    }
+}
+
+impl<T> FromGodot for *mut T {
+    fn try_from_godot(via: Self::Via) -> Option<Self> {
+        Some(via as Self)
+    }
+}
+
+impl<T> GodotCompatible for *const T {
+    type Via = i64;
+}
+
+impl<T> ToGodot for *const T {
+    fn to_godot(&self) -> Self::Via {
+        *self as i64
+    }
+}
+
+impl<T> FromGodot for *const T {
+    fn try_from_godot(via: Self::Via) -> Option<Self> {
+        Some(via as Self)
+    }
+}
+
+mod scalars {
+    use super::{impl_godot_as_self, FromGodot, GodotCompatible, ToGodot};
+    use crate::builtin::meta::GodotType;
+    use godot_ffi as sys;
+
```

Comment:
Not that it matters much, but I was always wondering if a separate `mod` here is the right approach, since it doesn't really add symbols in that module and creates extra indentation for all the code. 

To me, a horizontal delimiter and maybe 1 line comment immediately would probably be enough.